src/routes/customer-resource/customer-resource-show.js

```diff
@@ -1,53 +1,160 @@
+    this.context.addEpic('customerResourceShowLoad', ResourceLoadedEpic);
+
+    this.context.addReducer('customerResourceShow', handleActions({
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_LOAD]: (state, action) => {
+        let { vendorId, packageId, titleId } = action.params;
+        let endpoint = `${okapi.url}/eholdings/vendors/${vendorId}/packages/${packageId}/titles/${titleId}`;
+        return {...state, endpoint};
+      },
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_LOADED]: (state, action) => {
+        let resource = action.payload.customerResourcesList[0];
+
+        // slightly hacky, but we're flattening the customerResource
+        // onto the title to create a package title
+        let title = action.payload;
+        delete title.customerResourcesList;
+
+        return {
+          ...state,
+          isLoaded: true,
+          isErrored: false,
+          ...title,
+          ...resource
+        };
+      },
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED]: (state, action) => {
+        return {
+          ...state,
+          isSelected: !state.isSelected,
+          isTogglingSelection: true
+        };
+      },
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_RESOLVE]: (state, action) => {
+        return {
+          ...state,
+          isTogglingSelection: false
+        };
+      },
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_REJECT]: (state, action) => {
+        return {
+          ...state,
+          isTogglingSelection: false,
+          isSelected: !state.isSelected
+        };
+      },
+      [ActionTypes.CUSTOMER_RESOURCE_SHOW_ERROR]: (state, action) => {
+        return {
+          ...state,
+          toggleSelectionError: action.error
+        };
+      }
+    }, { isLoaded: false, isErrored: false }));
+
+    this.props.loadRecord(this.props.match.params);
+  }
 
   render() {
+    let { vendorId, packageId, titleId } = this.props.match.params;
     return (
-      <View customerResource={this.getCustomerResource()}/>
+      <View
+        model={this.props.model}
+        toggleSelected={this.props.toggleSelected}
+      />
     );
   }
+}
 
-  getCustomerResource() {
-    const {
-      resources: { showCustomerResource },
-      match: { params: { vendorId, packageId, titleId } }
-    } = this.props;
-
-    if (!showCustomerResource) {
-      return null;
+function mapStateToProps(state) {
+  let { okapi, customerResourceShow } = state;
+  return {
+    okapi,
+    model: customerResourceShow || {
+      isLoaded: false,
+      isErrored: false
     }
+  };
+}
 
-    return showCustomerResource.records.find((title) => {
-      return title.titleId == titleId && title.customerResourcesList.some((pkgTitle) => {
-        return pkgTitle.packageId == packageId && pkgTitle.vendorId == vendorId;
-      });
-    });
-  }
+// These are inline action creators
+const loadRecord = (params) => ({
```

Comment:
If the variables are not used more than once, Is there any reason not to inline them?

```js
function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    loadRecord(params){ 
      return { type: ActionTypes.CUSTOMER_RESOURCE_SHOW_LOAD, params };
    },
    toggleSelected() {
      return { type: ActionTypes.CUSTOMER_RESOURCE_TOGGLE_SELECTED };
    }
  }, dispatch);
}
```